src/gdbstub_impl/ext/host_io.rs

```diff
@@ -69,26 +69,15 @@ impl<T: Target, C: Connection> GdbStubImpl<T, C> {
                 HandlerStatus::Handled
             }
             HostIo::vFileFstat(cmd) if ops.enable_fstat().is_some() => {
-                let mut err: Result<_, Error<T::Error, C::Error>> = Ok(());
-                let mut callback = |data: &[u8]| {
-                    let e = (|| {
-                        res.write_str("F")?;
-                        res.write_num(data.len())?;
-                        res.write_str(";")?;
-                        res.write_binary(data)?;
-                        Ok(())
-                    })();
-
-                    if let Err(e) = e {
-                        err = Err(e)
-                    }
-                };
-
                 let ops = ops.enable_fstat().unwrap();
-                ops.fstat(cmd.fd, HostIoOutput::new(&mut callback))
-                    .handle_error()?;
-                err?;
-
+                let stat = ops.fstat(cmd.fd).handle_error()?;
+                let size = core::mem::size_of_val(&stat);
+                let p: *const HostStat = &stat;
+                let p: *const u8 = p as *const u8;
+                res.write_str("F")?;
+                res.write_num(size)?;
+                res.write_str(";")?;
+                res.write_binary(unsafe { core::slice::from_raw_parts(p, size) })?;
```

Comment:
ahhhhhh I know it's tempting to do this, but this is Rust, not C 😬 

`gdbstub` really tries to minimize the use of `unsafe` (see the README), and this is an operation that definately doesn't require the use of unsafe. 

While it'll be more verbose, you should manually call `write_binary` on each of the struct's fields. i.e: have a big block of `res.write_binary(&stat.{field}.to_le_bytes())?`. 

* * *

As an aside, just so you know, the way you've got it currently implemented results in UB. The `HostStat` struct isn't marked as `repr(C)`, which means we have no idea what the underlying in-memory representation is, so it's not safe to do this C-style struct-to-char* conversion.

In addition, it should be noted that this current implementation would be incorrect in the case where a little-endian host is providing a gdbstub for a big-endian target (or vice versa)

* * *

If this kind of thing starts cropping up in `gdbstub` more often, I'd consider pulling in some kind of library to help with this kind of struct-to-byte munging (e.g: `zerocopy`, `bytemuck`, etc...), but given that this is a one-off case, I'd rather not bring in those dependencies into the project just yet.